src/python/pants/help/help_printer.py

```diff
@@ -532,3 +542,41 @@ def _get_help_json(self) -> str:
         return json.dumps(
             self._all_help_info.asdict(), sort_keys=True, indent=2, cls=HelpJSONEncoder
         )
+
+    @staticmethod
+    def _get_json_schema(all_help_raw: str) -> Dict[str, Any]:
+        """Generate a JSON schema file contents (https://json-schema.org/).
+
+        This schema file can be used by IDEs (PyCharm, VSCode, etc) to provide autocompletion when
+        editing Pants configuration files in TOML format. It can also be used to validate the
+        configuration file(s) programmatically.
+        """
+        all_help = json.loads(all_help_raw)["scope_to_help_info"]
```

Comment:
It would make more sense to me if there was a `PantsOptionsJsonSchema` class (straw-man name for that class) in `help_json_schema.py` that takes an instance of `AllHelpInfo` as input to produce the schema output.

This json dumps/loads round-trip makes us loose type safety for one, and it would be nice to keep the schema logic in one place as much as possible :)

Reply:
I see what you mean. I am happy to move the code to `help_json_schema.py` from `help_printer.py`, that's easy.

> This json dumps/loads round-trip makes us loose type safety for one,
Yes, indeed. It does the necessary conversions for me of course, so that I could work with a simple dictionary when constructing a schema. However, I also get the point that accessing `self.all_help_info.scope_to_help_info` would be much better from type safety perspective. So perhaps you mean that I could stay in typed world when constructing the schema object and then dump it to JSON only at the return step.

I've experimented a bit and it requires knowing the function names and properties at runtime and using print statements won't be enough for iterative development. I'll write a dummy test to do debugging in IDE to explore the `AllHelpInfo` instance.

Reply:
Does it make sense?

Reply:
> So perhaps you mean that I could stay in typed world when constructing the schema object and then dump it to JSON only at the return step.

or as long as makes sense, yea. 

> [...] it requires knowing the function names and properties at runtime

I'm not sure I follow this..